Reader: Support Jetpack story block in shared css and open in a new tab on click

aforcier opened the pull request:
Alternative implementation to #44985 to support the mobile readers as well as Calypso. It relies on the shared reader css logic introduced in https://github.com/Automattic/wp-calypso/pull/43073.

#### Changes proposed in this Pull Request

@Tug we discussed this a bit and we're not big fans of duplicating the css from the block, but this seems to me the best way to support all the readers well.

I looked at doing things like `@import url("https://s0.wp.com/wp-content/mu-plugins/jetpack/_inc/blocks/story/view.css")` in the scss file itself, but this has two problems:

1. It includes quite a lot of unnecessary css only used for the fullscreen player (and not relevant to the reader).
2. It looks like the css isn't inlined when included from a URL, so instead another css file download is triggered. This isn't so bad for Calypso but the mobile readers rely on caching the reader-mobile.css, and the caching rules won't apply to that file.

2 surprised me a bit, when inlining a local file the css is rendered and included in the outputed reader-mobile.css, but apparently not for remote URLs. If there's a way to do that let me know!

As well as moving the CSS, I also moved the fullscreen/autoplay query parameters to the block renderer.

I did keep the `target=_blank` logic in Calypso because the mobile readers don't need it and it feels to me more like a reader concern. Happy to move that though if you think that makes more sense.

Note: Some small changes are needed in the mobile readers themselves. I'll open PRs for those too, but I'd like to know if this is headed in the right direction first.

<img width="1036" alt="Screen Shot 2020-09-04 at 10 47 53 AM" src="https://user-images.githubusercontent.com/9613966/92190435-2a49a600-ee9c-11ea-88c9-37b0050455b6.png">

#### Testing instructions

* Open the reader at http://calypso.localhost:3000/
* Subscribe to a site that has stories and has https://github.com/Automattic/jetpack/pull/17069 applied.
* Open a story in full page, confirm the CSS is applied and the preview looks like the screenshot.
* Click on the embed, a new tab should open and the story should load in fullscreen.

Tug commented on client/blocks/reader-full-post/_content.scss:
```diff
@@ -795,3 +795,8 @@
+// Import styles for Jetpack Story block.
+// This is a copy of Jetpack's extensions/blocks/story/player/style.scss, with
+// some unnecessary styles removed.
+@import 'story.scss';
```
I think it's best to keep the imports at the top of the file, for readability.

aforcier replied:
I was wondering about that - since it just inlines the css in the output I wasn't sure about putting stories at the top, but I'm happy to do it whichever way. Updated in a0f4fbc.

Tug reviewed (approved):
LGTM

aforcier commented:
@bluefuton could I trouble you for a review of this one from a Reader perspective please?

bluefuton commented:
> could I trouble you for a review of this one from a Reader perspective please?

Sure @aforcier :) Do you have a test site that has a few stories already?

aforcier commented:
> Sure @aforcier :) Do you have a test site that has a few stories already?

@bluefuton I'll message you 👍

bluefuton commented:
Looks solid to me 👍 

> we're not big fans of duplicating the css from the block

I also feel that way and would like an automated solution to bring Gutenberg block CSS over to Reader. In the meantime, this is the best approach we have.

I have some general notes about the accessibility of the markup - it that generated by Stories or have we tailored it for Reader? I notice for example there's no label/text alternative on the link, and `wp-story-embed-icon-expand` should be an interactive element like a button, not a div. Where's the appropriate place for that feedback?

<img width="283" alt="Screen Shot 2020-09-21 at 14 51 56" src="https://user-images.githubusercontent.com/17325/93729782-0adfa680-fc1a-11ea-9ac4-e750824fc1dd.png">

aforcier commented:
Thanks for looking @bluefuton!

> I have some general notes about the accessibility of the markup - it that generated by Stories or have we tailored it for Reader? I notice for example there's no label/text alternative on the link, and wp-story-embed-icon-expand should be an interactive element like a button, not a div. Where's the appropriate place for that feedback?

The markup is general to Stories and not tailored for the Reader. @Tug has some in-progress work on improving a11y for the block in https://github.com/Automattic/jetpack/pull/16852, which might end up addressing some of your feedback.

I'd suggest opening an issue on the Jetpack repo, if you tag myself and @Tug we'll combine it with the ongoing a11y work.
